sound/soc/sof/ipc4.c

```diff
@@ -599,6 +601,13 @@ static int ipc4_fw_ready(struct snd_sof_dev *sdev, struct sof_ipc4_msg *ipc4_msg
 	sdev->debug_box.offset = snd_sof_dsp_get_window_offset(sdev,
 							SOF_IPC4_DEBUG_WINDOW_IDX);
 
+	ret = snd_sof_dsp_block_read(sdev, SOF_FW_BLK_TYPE_SRAM, inbox_offset, fw_ready,
+				     sizeof(*fw_ready));
```

Comment:
`struct sof_ipc_fw_ready` is the IPC3 fw_ready message. IPC4 does not  define payload for FW_READY notification.
Reference fw definitely not giving any payload. The way to get the configuration and parameters is via GET_LARGE_CONFIG:FW_CONFIG, but that does not have a tuple defined for context save support as it is always supported in the reference fw along with IMR  boot.

Reply:
@ujfalusi why do we need to worry about the reference firmware in the context of library loading? Is this something we will ever test?

Reply:
We should only lose compatibility with a reference when all possible alternatives have been exhausted...